lib/fizzy/parser_expr.cpp

```diff
@@ -180,13 +215,19 @@ void push_branch_immediates(const ControlFrame& branch_frame, int stack_height,
     push(immediates, get_branch_arity(branch_frame));
 }
 
-inline void mark_frame_unreachable(ControlFrame& frame, Stack<ValType>& operand_stack) noexcept
+inline void mark_frame_unreachable(
+    ControlFrame& frame, Stack<OperandStackType>& operand_stack) noexcept
 {
     frame.unreachable = true;
     operand_stack.shrink(static_cast<size_t>(frame.parent_stack_height));
 }
 
-inline void push_operand(Stack<ValType>& operand_stack, ValType type)
+inline void push_operand(Stack<OperandStackType>& operand_stack, ValType type)
```

Comment:
Because of this overload, it may be better to just allow implicit conversion `ValType` -> `OperandStackType`.

Reply:
How would you do that exactly? I can imagine only making `OperandStackType` a class, containing another enum type, and implicit conversion constructor.

Reply:
Yes, it looks like. Not sure this is worth it then.

Reply:
Yeah, it looks that it would be quite verbose then.